Mochi/src/OrbitAI_Mochi.cpp

```diff
@@ -1,1424 +1,246 @@
-
+#include <cstdlib>      /* For exit() */
+#include <iostream>     /* For cout */
+#include <sys/types.h>  /* For mkdkir */
+#include <sys/stat.h>   /* For mkdkir */
+#include <unistd.h>
 #include <string>
-#include <sstream>
-#include <algorithm>
-#include <iterator>
-#include <ctime>
-
-// For machine learning.
-#include <mochimochi/binary_classifier.hpp>
-#include <mochimochi/utility.hpp>
-
-#define ENABLE_PA                                      0
+#include <map>
+#include <vector>
 
-#define COMMAND_BUFFER_LENGTH                        100
-#define SAVE_AFTER_UPDATE                              1    // Save/Serialize the models after each update.
-#define LOG_TIMES                                      1    // Include timestamps and execution times in the CSV log files.
+#include "Constants.hpp"
+#include "Utils.hpp"
+#include "PropertiesParser.hpp"
+#include "HyperParameters.hpp"
+#include "SocketServer.hpp"
+#include "MochiMochiProxy.hpp"
 
-#define LOG_FILEPATH_TRAINING        "logs/training.csv"
-#define LOG_FILEPATH_INFERENCE      "logs/inference.csv"
-#define LOG_FILEPATH_ORBITAI          "logs/orbitai.log"
+using namespace std;
 
-// In case no port number is given.
-#define DEFAULT_PORT_NUMBER                         9999
+/* Modes */
+enum class Mode {
+    trainNew = 0,
+    trainContinue = 1,
+    infer = 2
+};
 
-// Error codes.
-#define ERROR_SUCCESS                                  0
-#define ERROR_UNKNOWN                                  1
-#define ERROR_PARSE_ARGS                               2
-#define ERROR_INVALID_PORT_NUM                         3
-#define ERROR_CREATE_SOCKET                            4
-#define ERROR_BIND_PORT                                5
-#define ERROR_LISTEN_SOCKET                            6
-#define ERROR_GRAB_CONNECTION                          7
+/* Flag to keep track if models were loaded or not. */
+int gModelsLoadedFlag = 0;
 
 /**
- * Get a precise timestamp as a string.
+ * Process the received command.
+ * Returns flag indicating if the program loop should be exited or not.
  */
-std::string getTimestamp();
+int processReceivedCommand(int mode, int dim, string *pReceivedCommand, MochiMochiProxy *pMochiMochiProxy);
 
-/**
- * Logging into log file.
- */
-static inline void log(std::string message, std::string level);
-
-/**
- * Logging errors into the log file.
- */
-void logError(std::string message);
 
 /**
- * Logging info into the log file.
- */
-void logInfo(std::string message);
-
+ * Main application function.
+ */ 
+int main(int argc, char *argv[])
+{
+    if (argc != 2)
+    {
+        logError(ERROR_INVALID_ARGS, "invalid number or arguments given, expected a path to the properties file.");
+        exit(ERROR_INVALID_ARGS);
+    }
 
+    /* The vector that will contain points to the online ML algorithm class instances. */
+    /* We use a vector to preserve insertion order */
+    vector<pair<string, BinaryOMLCreator*>> bomlCreatorVector;

```

Comment:
I think you could let the proxy fully take care of this vector. You keep reference here only to destroy them at the end it seems.

Reply:
Good catch. I will fix.

Reply:
This is done. Ready for review.